File: python/program/Supplier.py
from Perusahaan import Perusahaan

# subclass Supplier
class Supplier(Perusahaan):

    def __init__(self, noHp: str, email: str, alamat: str,
                 nama: str, tahunBerdiri: int, bidangUsaha: str,
                 kodeSupplier: str, daftarProdukSupplier: str, lamaKerjasama: str):
        # panggil constructor parent (Perusahaan)
        super().__init__(noHp, email, alamat, nama, tahunBerdiri, bidangUsaha)
        # atribut khusus Supplier
        self._kodeSupplier = str(kodeSupplier)
        self._daftarProdukSupplier = str(daftarProdukSupplier)
        self._lamaKerjasama = str(lamaKerjasama)

    # Getter & Setter Kode Supplier
    def getKodeSupplier(self) -> str:
        return self._kodeSupplier

    def setKodeSupplier(self, kodeSupplier: str) -> None:
        self._kodeSupplier = str(kodeSupplier)

    # Getter & Setter Daftar Produk Supplier
    def getDaftarProdukSupplier(self) -> str:
        return self._daftarProdukSupplier

    def setDaftarProdukSupplier(self, daftarProdukSupplier: str) -> None:
        self._daftarProdukSupplier = str(daftarProdukSupplier)

    # Getter & Setter Lama Kerjasama
    def getLamaKerjasama(self) -> str:
        return self._lamaKerjasama

    def setLamaKerjasama(self, lamaKerjasama: str) -> None:
        self._lamaKerjasama = str(lamaKerjasama)

File: cpp/program/Supplier.cpp
#include <iostream>
#include "Perusahaan.h"
using namespace std;

/* Class Supplier yang mewarisi Perusahaan */
class Supplier : public Perusahaan{
    protected:
        string kodeSupplier;         // Kode unik supplier
        string daftarProdukSupplier; // Daftar produk yang disuplai
        string lamaKerjasama;        // Lama kerjasama (misal: "5 tahun")

    public:
        // Constructor kosong
        Supplier() : Perusahaan() {
            kodeSupplier = "";
            daftarProdukSupplier = "";
            lamaKerjasama = "";
        }

        // Constructor lengkap
        Supplier(string noHp, string email, string alamat,
                 string nama, int tahunBerdiri, string bidangUsaha,
                 string kodeSupplier, string daftarProdukSupplier, string lamaKerjasama)
                 : Perusahaan(noHp, email, alamat, nama, tahunBerdiri, bidangUsaha) {
            this->kodeSupplier = kodeSupplier;
            this->daftarProdukSupplier = daftarProdukSupplier;
            this->lamaKerjasama = lamaKerjasama;
        }

        // Getter dan Setter
        string getKodeSupplier() {
            return kodeSupplier;
        }

        void setKodeSupplier(string kodeSupplier) {
            this->kodeSupplier = kodeSupplier;
        }

        string getDaftarProdukSupplier() {
            return daftarProdukSupplier;
        }

        void setDaftarProdukSupplier(string daftarProdukSupplier) {
            this->daftarProdukSupplier = daftarProdukSupplier;
        }

        string getLamaKerjasama() {
            return lamaKerjasama;
        }

        void setLamaKerjasama(string lamaKerjasama) {
            this->lamaKerjasama = lamaKerjasama;
        }
        ~Supplier(){}
};
